src/benchmarks/src/generic_bench.cpp

```diff
@@ -0,0 +1,251 @@
+  int num_batches = get_int(NUM_BATCHES, config);
+  int request_batch_size = get_int(REQUEST_BATCH_SIZE, config);
+  long batch_delay_micros = get_long(REQUEST_BATCH_DELAY_MICROS, config);
+  int latency_objective = get_int(LATENCY_OBJECTIVE, config);
+  std::string model_name = get_str(MODEL_NAME, config);
+  std::string model_version = get_str(MODEL_VERSION, config);
+  bool draw_from_poisson = get_bool(POISSON_DELAY, config);
+  bool prevent_cache_hits = get_bool(PREVENT_CACHE_HITS, config);
+
+  int num_datapoints = static_cast<int>(data.size());
+  std::vector<double> query_vec;
+  std::default_random_engine generator;
+  std::poisson_distribution<int> distribution(batch_delay_micros);
+  long delay_micros;
+  int data_index;
+  double label;
+  std::vector<double> input_vector_data;
+
+  for (int j = 0; j < num_batches; j++) {
+    std::vector<boost::future<Response>> futures;
+    std::vector<double> future_labels;
+    for (int i = 0; i < request_batch_size; i++) {
+      // Select datapoint
+      data_index = indices[(j * request_batch_size + i) % num_datapoints];
+      query_vec = data[data_index];
+      label = labels[data_index];
+
+      if (prevent_cache_hits) {
+        // Modify it to be epoch and thread-specific
+        query_vec[0] = (j * request_batch_size + i) / num_datapoints;
+        query_vec[1] = thread_id;
+      }
+
+      // Copy the datapoint into a new shared pointer
+      std::shared_ptr<Input> input = std::make_shared<DoubleVector>(query_vec);
+
+      boost::future<Response> prediction =
+          qp.predict({TEST_APPLICATION_LABEL,
+                      UID,
+                      input,
+                      latency_objective,
+                      clipper::DefaultOutputSelectionPolicy::get_name(),
+                      {VersionedModelId(model_name, model_version)}});
+
+      futures.push_back(std::move(prediction));
+      future_labels.push_back(std::move(label));
+      bench_metrics.send_rate_->mark(1);
+    }
+
+    std::shared_ptr<std::atomic_int> completed =
+        std::make_shared<std::atomic_int>(0);
+    std::pair<boost::future<void>, std::vector<boost::future<Response>>>
+        results = future::when_all(std::move(futures), completed);
+    results.first.get();
+    for (int i = 0; i < static_cast<int>(results.second.size()); i++) {
+      boost::future<Response> &f = results.second[i];
+      Response r = f.get();
+
+      // Update metrics
+      if (r.output_is_default_) {
+        bench_metrics.default_pred_ratio_->increment(1, 1);
+      } else {
+        bench_metrics.default_pred_ratio_->increment(0, 1);
+      }
+      bench_metrics.latency_->insert(r.duration_micros_);
+      bench_metrics.num_predictions_->increment(1);
+      bench_metrics.throughput_->mark(1);
+
+      if (std::stod(r.output_.y_hat_) == future_labels[i]) {
+        bench_metrics.accuracy_ratio_->increment(1, 1);
+      } else {
+        bench_metrics.accuracy_ratio_->increment(0, 1);
+      }
```

Comment:
I agree that measuring accuracy here is besides the point and complicated by us changing feature vectors. That said, the old `end_to_end_bench` measured accuracy. Should we still include a way to run accuracy checks? I don't see a great reason to.